crates/proof-impl/cl-stf/src/lib.rs

```diff
@@ -40,25 +48,57 @@ fn verify_l2_block(
     );
 
     // Verify proof public params matches the exec segment
-    let proof_exec_segment = reconstruct_exec_segment(el_proof_pp);
+    let (proof_exec_segment, el_deposit_data) = reconstruct_exec_segment(el_proof_pp);
     let block_exec_segment = block.body().exec_segment().clone();
     assert_eq!(proof_exec_segment, block_exec_segment);
+
+    el_deposit_data
 }
 
 /// Generates an execution segment from the given ELProof public parameters.
-pub fn reconstruct_exec_segment(el_proof_pp: &ELProofPublicParams) -> ExecSegment {
-    // create_evm_extra_payload
+pub fn reconstruct_exec_segment(
+    el_proof_pp: &ELProofPublicParams,
+) -> (ExecSegment, Vec<ELDepositData>) {
+    let withdrawals = el_proof_pp
+        .withdrawal_intents
+        .iter()
+        .map(|intent| {
+            bridge_ops::WithdrawalIntent::new(
+                BitcoinAmount::from_sat(intent.amt),
+                XOnlyPk::new(Buf32(intent.dest_pk)),
+            )
+        })
+        .collect::<Vec<_>>();
+
+    let el_deposit_data: Vec<ELDepositData> = el_proof_pp
+        .deposit_requests
+        .iter()
+        .map(|deposit_request| {
+            ELDepositData::new(
+                deposit_request.index,
+                gwei_to_sats(deposit_request.amount),
+                deposit_request.address.as_slice().to_vec(),
+            )
+        })
+        .collect();
+
+    let applied_ops = el_deposit_data
+        .iter()
+        .map(|deposit_data| Op::Deposit(deposit_data.clone()))
+        .collect();
```

Comment:
Try to avoid using anything with this `Op` type.  The deposit intents queue is the source of truth.  See other comment.